presto-router/src/main/java/com/facebook/presto/router/scheduler/UserHashScheduler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.router.scheduler;
+
+import com.facebook.airlift.log.Logger;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+public class UserHashScheduler
+        implements Scheduler
+{
+    private List<URI> candidates;
+
+    private static final Logger log = Logger.get(UserHashScheduler.class);
+
+    @Override
+    public Optional<URI> getDestination(String user)
+    {
+        try {
+            return Optional.ofNullable(candidates.get(user.hashCode() % candidates.size()));
+        }
+        catch (Exception e) {
+            log.warn(e, "Error getting destination for user ", user);
+            return Optional.empty();
+        }
```

Comment:
very similar as the random scheduler, I don't think the ofNullable is necessary, we could use `of` instead.  Also, the cache exception is not necessary. what do you think?

Reply:
Thanks for the suggestion! Similar to that in the random choice scheduler, I updated the `Optional.ofNullable` to `Optional.of` and updated the general exception with the concrete exception.